nextflow_aws_batch_trigger.py

```diff
@@ -0,0 +1,314 @@
+            + "has been created."
+        )
+    else:
+        logging.info(
+            f"{now.strftime('%Y-%m-%d %H:%M:%S')} The Queue : {job_queue_name} has already been created. "
+            f"Hence, moving to the next process"
+        )
+
+
+def create_instance(
+    ec2_client,
+    key_name,
+    instance_role,
+    s3_data,
+    s3_bucket,
+    result_location,
+    script_name,
+    config_file_name,
+    s3_logging_dir,
+    s3_result,
+    subnet1,
+    security_groupId,
+):
+    instances = ec2_client.run_instances(
+        ImageId="ami-0b0dcb5067f052a63",
+        MinCount=1,
+        MaxCount=1,
+        InstanceType="t2.micro",
+        KeyName=key_name,
+        IamInstanceProfile={"Name": instance_role},
+        UserData="""#!/bin/bash 
+                     sudo rpm --import https://yum.corretto.aws/corretto.key 
+                     sudo curl -L -o /etc/yum.repos.d/corretto.repo https://yum.corretto.aws/corretto.repo 
+                     sudo yum install -y  java-11-amazon-corretto-devel 
+                     sudo amazon-linux-extras install epel -y
+                     sudo yum install s3fs-fuse -y 
+                     aws s3 sync """
+        + s3_data
+        + """ /home/ec2-user/
+                     sudo chmod 777 /home/ec2-user/nextflow 
+                     cd /home/ec2-user
+                     mkdir """
+        + result_location
+        + """ sudo s3fs """
+        + s3_bucket
+        + """ """
+        + result_location
+        + """ -o allow_other -o umask=000 -o iam_role=auto 
+                     ./nextflow run /home/ec2-user/"""
+        + script_name
+        + """ -c """
+        + config_file_name
+        + """ -bucket-dir """
+        + s3_logging_dir
+        + """ --outdir="""
+        + s3_result
+        + """ cd /home/ec2-user/ 
+                     touch done.txt
+                     aws s3 cp /home/ec2-user/done.txt """
+        "s3://" + s3_bucket + "/" + result_location,
+        NetworkInterfaces=[
+            {
+                "AssociatePublicIpAddress": True,
+                "DeleteOnTermination": True,
+                "DeviceIndex": 0,
+                "SubnetId": subnet1,
+                "Groups": [security_groupId],
+            },
+        ],
+    )
+    logging.info(
+        f"{now.strftime('%Y-%m-%d %H:%M:%S')} The instance: {instances['Instances'][0]['InstanceId']} has been created and the nextflow script {script_name} has been triggered"
+    )
+
+    return instances["Instances"][0]["InstanceId"]
+
+
+def check_result(s3, s3_bucket, id_instance, result_location):
+    bucket_name = s3_bucket
+    file_path = result_location + "done.txt"
+
+    while True:
+        try:
+            s3.head_object(Bucket=bucket_name, Key=file_path)
+        except:
+            logging.info(
+                f"{now.strftime('%Y-%m-%d %H:%M:%S')} File does not exist in S3 bucket. Checking again in 5 mins."
+            )
+
+            time.sleep(300)
```

Comment:
I think we should add a timeout after a certain amount of time here. Else if the process locks up then we wouldn't want the process to continue.

Reply:
Timeout has been added in the function, can you please review.